src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -63,7 +67,14 @@ case class CsvRelation protected[spark] (
 
   // By making this a lazy val we keep the RDD around, amortizing the cost of locating splits.
   def buildScan = {
-    val baseRDD = sqlContext.sparkContext.textFile(location)
+    val baseRDD = if (charset == DefaultCharset) {
```

Comment:
Would you please factor out this as a utility function that takes sparkContext and the charset and returns the RDD? Then you can reuse it in line 140 to get first line.